.github/workflows/tests.yml

```diff
@@ -4,36 +4,69 @@ on: [push, pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macOS-latest]
+        deno-version: [1.19, 1.17, 1.15]
+
     steps:
     - name: Install Deno
-      run: curl -fsSL https://deno.land/x/install/install.sh | sh
+      uses: denoland/setup-deno@v1
+      with:
+        deno-version: ${{ matrix.deno-version }}
+
     - uses: actions/checkout@v1
     - name: Run the formatter
       run: |
-        shopt -s globstar failglob
-        /home/runner/.deno/bin/deno fmt --check
+        deno fmt --check
+
     - name: Build SQLite
       id: build
+      if: ${{ matrix.os != 'windows-latest' }}
       run: |
         export DENO_INSTALL="/home/runner/.deno"
         export PATH="$DENO_INSTALL/bin:$PATH"
-        sudo apt-get -y install --no-install-recommends tclsh gcc brotli
         cd build
         make setup amalgamation release
+
+    - name: Setup MinGW environment
+      if: ${{ matrix.os == 'windows-latest' }}
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: MINGW64
+        update: true
+        install: git gcc make mingw-w64-x86_64-cc
+
+    - name: Build SQLite (Windows)
```

Comment:
We don't need this step to be separate from the linux build, since `setup-deno` sets up the path for us.

So can delete the `export DENO_INSTALL`/`export PATH` statements from the command, delete the windows specific version, and move the `Setup MinGW Environment` step before the build step.